.pre-commit-config.yaml

```diff
@@ -0,0 +1,20 @@
+# Usage: install pre-commit, and then run `pre-commit install` to install git hooks
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+    -   id: end-of-file-fixer
+    -   id: trailing-whitespace
+-   repo: local
+    hooks:
+    -   id: rustfmt
+        name: rustfmt
+        entry: rustfmt
```

Comment:
- `rustfmt` vs `cargo fmt`?
- `fmt` vs `fmt --check`?

Reply:
It will run against diff, so rustfmt. And it is silly to check but not fix.